pkg/server/config/config_store.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2024 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/go-github/v64/github"
+	"gopkg.in/yaml.v3"
+
+	"github.com/abcxyz/pkg/cache"
+	"github.com/abcxyz/pkg/githubauth"
+)
+
+const assertionKey string = "assertion"
+
+// GitHubClientProvider is a function that given an org and repo
+// will return a properly configured GitHub client object that
+// can be used to make API calls to GitHub.
+type GitHubClientProvider func(ctx context.Context, org, repo string) (*github.Client, error)
+
+// ConfigStore is a container for Config data that can
+// be loaded from various sources. The Config file contents
+// are retrieved, parsed and then cached before being returned.
+// Config objects expire after the specified amount of time and
+// are re-retrieved.
+type ConfigStore struct {
+	loader configFileLoader
+	cache  *cache.Cache[*Config]
+}
+
+// NewConfigStore creates a ConfigStore object with the specfied cache
+// expiration time, local configuration directory and a GitHub App for
+// authenticating GitHub requests.
+func NewConfigStore(expireAfter time.Duration, localConfigDir string, app *githubauth.App) *ConfigStore {
+	// create and configure all of the file loaders
+	localLoader := localConfigFileLoader{configDir: localConfigDir}
+	inRepoLoader := ghInRepoConfigFileLoader{
+		provider: func(ctx context.Context, org, repo string) (*github.Client, error) {
+			return makeGitHubClient(ctx, app, org, repo)
+		},
+		configPath: ".github/minty.yaml",
+		ref:        "main",
+	}
+	orgLoader := fixedRepoConfigFileLoader{
+		repo: ".google-github",
```

Comment:
Should this be configurable given this an oss project?

Reply:
Yes, that will be passed in when I integrate this with the rest of the code. I just hard coded it for now so you knew what the intended values are.